cni/network/network_windows.go

```diff
@@ -266,57 +265,61 @@ func getPoliciesFromRuntimeCfg(nwCfg *cni.NetworkConfig, isIPv6Enabled bool) []p
 
 		// To support hostport policy mapping
 		// uint32 NatFlagsLocalRoutedVip = 1
-		rawPolicy, _ := json.Marshal(&hnsv2.PortMappingPolicySetting{
+		// To support hostport policy mapping for ipv6 in dualstack overlay mode
+		// uint32 NatFlagsIPv6 = 2
+
+		var flag hnsv2.NatFlags
+		if mapping.HostIp != "" {
+			hostIP, err := netip.ParseAddr(mapping.HostIp)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to parse hostIP %v", hostIP)
+			}
+
+			switch {
+			case hostIP.Is4():
+				flag = hnsv2.NatFlagsLocalRoutedVip
+			case hostIP.Is6() && isIPv6Enabled:
+				flag = hnsv2.NatFlagsIPv6
+			}
+		} else {
+			flag = hnsv2.NatFlagsLocalRoutedVip
+			if isIPv6Enabled {
+				flag = hnsv2.NatFlagsIPv6
```

Comment:
if `hostIP` is empty, shouldn't we just assume ipv4? How is `isIPv6Enabled` determined?

Reply:
What happens, when hostIp is v6 but ipv6Enabled is `false` ? Should that error our invalid hostIp ?

Reply:
If customer provides wrong ipv6 address in v4Overlay cluster which ipv6Enabled is false, then it will pass the wrong flag to policy but customer cannot create a valid pod. So this is controlled by customer and we cannot return error here